Easier code to manage + port values to new primitives

maximedegreve opened the pull request:
The `theme.js` file had a lot of repetition and started to become slightly unreadable because of the ever growing themes.

Before we use the primitives I now convert them all to hex values with alpha support first. That way it's less prone to mistakes if we forget to convert them. The `hex(color)` format is therefore not needed anymore.

I've also introduced a few helper methods to make the code look less heavy:

- **onlyDark(color)**: Target only dark modes without having to write the whole dictionary. This should scale better when new themes are introduced.
- **lightDark(lightColor, darkColor)**: Target all light and dark modes without having to write the whole dictionary. This should scale better when new themes are introduced.
- **alpha(color, alpha)**: Adds/changes the alpha value of a hex color.
- **onlyDarkHighContrast(color)**: Needs often separate adjustments so I've added this one as well.

I've also started with moving hardcoded values to our primitives as well. I need to still test them all throughout.

maximedegreve commented on src/theme.js:
```diff
@@ -1,243 +1,252 @@
+      "editor.inactiveSelectionBackground"    : color.accent.subtle,
+      "editor.selectionBackground"            : color.accent.muted,
+      "editor.selectionHighlightBackground"   : color.success.muted,
+      "editor.selectionHighlightBorder"       : null, // I need to figure this out as I think this was nothing....
```
@simurai Do you know what's happening here? I think those were just transparent.

simurai replied:
Hmm.. right. These seem to be `transparent`. Maybe they are just a left over from playing around and then not needing them or intentionally setting them to transparent? I guess it could also just be `#00000000`. Will `null` render as transparent too?

simurai commented on src/theme.js:
```diff
@@ -1,243 +1,252 @@
+      "editor.foreground"                 : color.fg.default,
+      "editor.background"                 : color.canvas.default,
+      "editorWidget.background"           : color.canvas.overlay,
+      "editor.foldBackground"             : color.neutral.muted,
```
Here I'm not sure. I think this color needs to be semi-transparent otherwise it would cover something else. As of today `color.neutral.muted` is semi-transparent and should work, but that might change in the future. So I thought it's safer to add the semi-transparency here in the VS Code theme.

maximedegreve replied:
I've reverted those to ones with alpha for safety

simurai commented on src/theme.js:
```diff
@@ -1,243 +1,252 @@
+      "editorWhitespace.foreground"       : color.fg.subtle,
+      "editorCursor.foreground"           : color.accent.fg,
+
+      "editor.findMatchBackground"            : color.attention.muted,
```
Same as above, I wonder if it's safer to use something like `alpha(color.attention.emphasis, 0.3)` to make sure they always stay semi-transparent even if Primitives changes. That said, I don't expect these global variables to change much, so maybe fine? 🤔

maximedegreve replied:
I've reverted those back to alphas.

simurai reviewed (commented):
Nice. ✨  Much simpler!

Added some comments about keeping the transparency in this theme and not rely on Primitives for that. Since it's more a happy coincidence. But we could also give it a try as is. Maybe then we could still add comments like `// needs transparency` so we can keep that in mind.

maximedegreve commented:
Made a few updates that addressed the above comments...

### Brackets
<img width="924" alt="Screenshot 2022-01-24 at 14 34 06" src="https://user-images.githubusercontent.com/980622/150802505-b18e511b-4d25-4c1a-9d96-ef687ebac69f.png">

It seems like `null` gives it a default grey outline this explains why we had a full alpha value there. I've created a `transparent` helper variable instead to make this more clear in the future.

### Word highlight
<img width="1629" alt="Screenshot 2022-01-24 at 14 28 32" src="https://user-images.githubusercontent.com/980622/150801569-fe192937-2b4e-473e-934e-add039680f7e.png">

In the current live version when putting your cursor inside a word the word gets a green outline. This doesn't seem right and instead I've replaced it all with our neutral colors.
